### ISSUE TITLE
Copying geometry objects

### ISSUE THREAD

**Issue by hiptmair** [craffael/lehrfempp#23]
During mesh generation it could happen that the geometry of all edges of a cell is known, but the geometry of the cell itself has not been supplied. 

The following options are available
* use isoparametric shape for cell by simply querying the position of a few points on the edges.
* use exact blending: in this case the parameterization of the edges has to be incorporated into the geometry object of the cell, which entails cloning the edge geometry. 

The second option can probably rely on the `SubGeometry()` method with a zero codimension argument, can't it?

**Comment by craffael** [comment]
Yes, I think we can "reuse" the SubGeometry() method to clone a Geometry object.

If, during mesh Generation it happens that the geometry of a cell must be constructed from the edges, I suggest that this is NOT handled by the MeshFactory because there are in principle many different ways for how the cell-geometry can be constructed from the edge geometries. If the  user added some edge entities to a mesh factory (via `AddEntity()`) without specifying the corresponding cell entities, then the MeshFactory should throw an error when the `build()` method is invoked.
So the user is responsible for adding the appropriate cell entities to the MeshFactory with their geometry.
